tests/integration/TestTransaction.cc

```diff
@@ -0,0 +1,322 @@
+
+    {
+        std::cout << "========== Producer: commitTransaction, Consumer: isolation.level=read_committed ==========" << std::endl;
+
+        const Topic     topic     = Utility::getRandomString();
+        KafkaTestUtility::CreateKafkaTopic(topic, 1, 3);
+
+        sendMessageWithTransactions("message to commitTransaction", topic, TransactionClosureAction::ToComplete);
+
+        const auto messages = receiveMessages(topic, IsolationLevel::ReadCommitted);
+        ASSERT_EQ(1, messages.size());
+    }
+
+    {
+        std::cout << "========== Producer: commitTransaction, Consumer: isolation.level=read_uncommitted ==========" << std::endl;
+
+        const Topic     topic     = Utility::getRandomString();
+        KafkaTestUtility::CreateKafkaTopic(topic, 1, 3);
+
+        sendMessageWithTransactions("message to commitTransaction", topic, TransactionClosureAction::ToComplete);
+
+        const auto messages = receiveMessages(topic, IsolationLevel::ReadUnCommitted);
+        ASSERT_EQ(1, messages.size());
+    }
+
+    {
+        std::cout << "========== Producer: abortTransaction, Consumer: isolation.level=read_committed ==========" << std::endl;
+
+        const Topic     topic     = Utility::getRandomString();
+        KafkaTestUtility::CreateKafkaTopic(topic, 1, 3);
+
+        sendMessageWithTransactions("message to abortTransaction", topic, TransactionClosureAction::ToAbort);
+
+        const auto messages = receiveMessages(topic, IsolationLevel::ReadCommitted);
+        ASSERT_EQ(0, messages.size());
+    }
+
+    {
+        std::cout << "========== Producer: abortTransaction, Consumer: isolation.level=unread_committed ==========" << std::endl;
+
+        const Topic     topic     = Utility::getRandomString();
+        KafkaTestUtility::CreateKafkaTopic(topic, 1, 3);
+
+        sendMessageWithTransactions("message to abortTransaction", topic, TransactionClosureAction::ToAbort);
+
+        const auto messages = receiveMessages(topic, IsolationLevel::ReadUnCommitted);
+        ASSERT_EQ(0, messages.size());
+    }
+
+    {
+        std::cout << "========== Producer: no commit/abortTransaction, Consumer: isolation.level=read_committed ==========" << std::endl;
+
+        const Topic     topic     = Utility::getRandomString();
+        KafkaTestUtility::CreateKafkaTopic(topic, 1, 3);
+
+        sendMessageWithTransactions("message with no commit/abortTransaction", topic, TransactionClosureAction::ToAbort);
+
+        const auto messages = receiveMessages(topic, IsolationLevel::ReadCommitted);
+        ASSERT_EQ(0, messages.size());
+    }
+
+    {
+        std::cout << "========== Producer: no commti/abortTransaction, Consumer: isolation.level=unread_committed ==========" << std::endl;
+
+        const Topic     topic     = Utility::getRandomString();
+        KafkaTestUtility::CreateKafkaTopic(topic, 1, 3);
+
+        sendMessageWithTransactions("message with no commit/abortTransaction", topic, TransactionClosureAction::ToAbort);
```

Comment:
Should this be NoAction instead ToAbort?

Reply:
`ToAbort` means `producer.abortTransaction();` would be called later, while `NoAction` means do nothing (doesn't commitTransaction, nor abortTransaction)

Reply:
According to the above message "========== Producer: no commti/abortTransaction, Consumer: isolation.level=unread_committed ==========", did you plan to test "NoAction" and "Read_UnCommitted"?

Reply:
Oh, I see, yeah, here should test the `NoAction` case, while I missed to change the parameter for the case. Thanks! 😅